### DISCUSSION TITLE
Auto index a specific folder on file changes using docker

### DISCUSSION THREAD

**Discussion by whoshoe** [photoprism/photoprism#2202] [General]
Hello guys, 

I just started using Photoprism recently and I couldn't be happier with it. Since I use nextcloud for photo uploads from my iPhones and iPads the auto index doesn't work seamlessly. I tried using a cron job to index the nextcloud folder every 30 mins but most of the time there's nothing to index. After banging my head against walls for a few days I finally found a solution which allows me to watch for file changes in real time and run a custom script when changes are detected. It even has a cool down counter. What's more important to me is that I can run everything in a container. I'd like to share what I found and let more people know this tool. 

The docker I am using is https://hub.docker.com/r/coppit/inotify-command. It uses inotify API to monitor file system events. There's sample configurations and detailed documentation for parameters on the page. The command for indexing is pretty standard: `docker exec photoprism photoprism index /nextcloud`.

Becase we don't have access to the `docker exec` command inside a container, we need to do the following mappings:
1. map the host docker socket to the container: `-v /var/run/docker.sock:/var/run/docker.sock`
2. map the host docker bin file to the container: `-v /usr/bin/docker:/usr/bin/docker`

Docker run the container and enjoy!

**Comment by scylla-xi** [comment]
thanks for sharing!

does inotify monitor changes in subdirectories?
e.g.
* I mount dir `/dir1` to `inotify-command`
* I add new file in `/dir1/subdir/`
* will the container execute the `index` command?

EDIT: 
I found some time to play around and test `inotify-command`
now I can answer the question myself 🤗
all subdirectories are monitored 👍

**Reply by nanonerd-io** [reply]
Hello whoshoe

I know this post is from several months ago, but I'm literally trying to accomplish the exact same thing (using Nextcloud to auto upload photos to the InstantUpload folder from my Android device). I've setup Photoprism via docker-compose and it worked in that it indexed that folder, but now I have new photos that I'd like Photoprism to import/add without re-indexing the entire library. Would you mind sharing some more details / specifics about how you set this up? I'm a bit confused and don't know where to start. Any advice would be greatly appreciated as this is the missing piece of my HTPC setup

Thanks !!

**Reply by lastzero** [reply]
You can easily index a specific directory by specifying it as an argument in the terminal or by selecting it from the drop-down menu in the user interface. It is also possible to update only new and changed files by omitting "Complete Rescan". To monitor and automatically index a folder without using WebDAV, you would need to write a short script that triggers this until we have a built-in solution.

**Comment by Floppik** [comment]
> Hello guys,
> 
> I just started using Photoprism recently and I couldn't be happier with it. Since I use nextcloud for photo uploads from my iPhones and iPads the auto index doesn't work seamlessly. I tried using a cron job to index the nextcloud folder every 30 mins but most of the time there's nothing to index. After banging my head against walls for a few days I finally found a solution which allows me to watch for file changes in real time and run a custom script when changes are detected. It even has a cool down counter. What's more important to me is that I can run everything in a container. I'd like to share what I found and let more people know this tool.
> 
> The docker I am using is https://hub.docker.com/r/coppit/inotify-command. It uses inotify API to monitor file system events. There's sample configurations and detailed documentation for parameters on the page. The command for indexing is pretty standard: `docker exec photoprism photoprism index /nextcloud`.
> 
> Becase we don't have access to the `docker exec` command inside a container, we need to do the following mappings:
> 
> 1. map the host docker socket to the container: `-v /var/run/docker.sock:/var/run/docker.sock`
> 2. map the host docker bin file to the container: `-v /usr/bin/docker:/usr/bin/docker`
> 
> Docker run the container and enjoy!

Hi! Is map socket and bin file enough? If i try execute docker command, i get this error:
bash: /usr/bin/docker: No such file or directory

```
e4e98c622003cf83ddbf85527f322b014dba8767c0b5b19965bb44d2263e64be
AppArmorProfile docker-default
Args [ ]
Config { AttachStderr: false, AttachStdin: false, AttachStdout: false, Cmd: /sbin/boot.sh, Domainname: , Entrypoint: null, Env: PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin,TERM=xterm-256color,LANG=en_US.UTF-8,LANGUAGE=en_US:en,LC_ALL=en_US.UTF-8,UMAP=,GMAP=, Hostname: 0308d725d16f, Image: coppit/inotify-command:latest, Labels: [object Object], OnBuild: null, OpenStdin: false, StdinOnce: false, Tty: false, User: , Volumes: [object Object], WorkingDir: }
Created 2022-10-03T11:34:30.805859091Z
Driver overlay2
ExecIDs [ 42cccc6689f4baaffa09bcf7551d3d856393087d8e34800cc25e46b122bdd554 ]
GraphDriver { Data: [object Object], Name: overlay2 }
HostConfig { AutoRemove: false, Binds: /mnt/Toshiba/nc_data/admin/files/Фото/Автозагрузка/2022:/dir1,5e3889cc7aa5bcc2a29b3badea6ea23413f1037f169506a605758298ed8656c0:/dir7,45d99435e3c2176958fafa507e59ccb78697049b4c5e818ad0dbe7f5cb8005fb:/dir12,635c2fe987aaa6b276b89bd667f58675cdebf832404a4808ed7bebd3bebf5a60:/dir15,/etc/localtime:/etc/localtime,cfbeff9562a7b097bccbb4a4871e78b68c779dc5fdf1356b4106c0d6fbeead18:/dir11,/home/floppik/inotify-config:/config,758c785b5e42e41284fc67b584e43ebde540c7712b4a511b3876b4151c257890:/dir8,7b9acb904058c50412b6dd4179177333407e652d24ac5e2bd80a123a80b11f64:/dir20,57d9a17b9bbd1e63425542bd1113da7db0ba2f1077a376c597a6b929f2712292:/dir2,cc5c474dc43718f237239ef7b5b3c9f0b8185030a1e4358eeb8dcdb9194fd968:/dir4,f74f0a8fd52790cc435242d6851de01f6402023536cde568733e73b818b8e44a:/dir6,3ef3c20c0a61521fc1e3d767d7896b7c3c8c7a0a01d4cdae0289ebfa52691663:/dir17,1531d2dc17d569157833636302dd2ecb049270300bfba36e15ca429e5ded3893:/dir13,d0456f96284f3147e785df9c7cbcc2e898a72f7284da18d21a6d3582c8e6e9f1:/dir9,9a72376441359f621fe9dbd0438607a6350376524b94c9a40ebcafcdc8714f00:/dir5,180f3c1164ab21aa4a49b19543e9611f21de65ec36eb28d01a045f59eaeb7aea:/dir18,baa10d52f872581f8251a6f5e2452075e922620b546d28741b960e74565ae842:/dir16,/var/run/docker.sock:/var/run/docker.sock,b09dd335db93f0cd57531e22f750e67f3a920378c056a4c4593692e6ab1626a9:/dir10,64bd97a71fa6b083b77ed1074303da117449e6e1b71dfbeec9eaf70b912630cf:/dir19,a1bfe0b4fc6f821efd6db3646db1a8ef1c52fe16885feefb05a970673cd71dbf:/dir14,77d50377090096c9226d743f031ac9023285551bd3aef706aa3ac4c3f55ade6c:/dir3,/usr/bin/docker:/usr/bin/docker, BlkioDeviceReadBps: null, BlkioDeviceReadIOps: null, BlkioDeviceWriteBps: null, BlkioDeviceWriteIOps: null, BlkioWeight: 0, BlkioWeightDevice: , CapAdd: AUDIT_WRITE,CHOWN,DAC_OVERRIDE,FOWNER,FSETID,KILL,MKNOD,NET_BIND_SERVICE,NET_RAW,SETFCAP,SETGID,SETPCAP,SETUID,SYS_CHROOT, CapDrop: AUDIT_CONTROL,BLOCK_SUSPEND,DAC_READ_SEARCH,IPC_LOCK,IPC_OWNER,LEASE,LINUX_IMMUTABLE,MAC_ADMIN,MAC_OVERRIDE,NET_ADMIN,NET_BROADCAST,SYSLOG,SYS_ADMIN,SYS_BOOT,SYS_MODULE,SYS_NICE,SYS_PACCT,SYS_PTRACE,SYS_RAWIO,SYS_RESOURCE,SYS_TIME,SYS_TTY_CONFIG,WAKE_ALARM, Cgroup: , CgroupParent: , CgroupnsMode: private, ConsoleSize: 0,0, ContainerIDFile: , CpuCount: 0, CpuPercent: 0, CpuPeriod: 0, CpuQuota: 0, CpuRealtimePeriod: 0, CpuRealtimeRuntime: 0, CpuShares: 0, CpusetCpus: , CpusetMems: , DeviceCgroupRules: null, DeviceRequests: null, Devices: , Dns: , DnsOptions: , DnsSearch: , ExtraHosts: , GroupAdd: null, IOMaximumBandwidth: 0, IOMaximumIOps: 0, IpcMode: private, Isolation: , KernelMemory: 0, KernelMemoryTCP: 0, Links: null, LogConfig: [object Object], MaskedPaths: /proc/asound,/proc/acpi,/proc/kcore,/proc/keys,/proc/latency_stats,/proc/timer_list,/proc/timer_stats,/proc/sched_debug,/proc/scsi,/sys/firmware, Memory: 0, MemoryReservation: 0, MemorySwap: 0, MemorySwappiness: null, NanoCpus: 0, NetworkMode: bridge, OomKillDisable: null, OomScoreAdj: 0, PidMode: , PidsLimit: null, PortBindings: [object Object], Privileged: false, PublishAllPorts: false, ReadonlyPaths: /proc/bus,/proc/fs,/proc/irq,/proc/sys,/proc/sysrq-trigger, ReadonlyRootfs: false, RestartPolicy: [object Object], Runtime: runc, SecurityOpt: null, ShmSize: 67108864, UTSMode: , Ulimits: null, UsernsMode: , VolumeDriver: , VolumesFrom: null }
HostnamePath /var/lib/docker/containers/e4e98c622003cf83ddbf85527f322b014dba8767c0b5b19965bb44d2263e64be/hostname
HostsPath /var/lib/docker/containers/e4e98c622003cf83ddbf85527f322b014dba8767c0b5b19965bb44d2263e64be/hosts
Id e4e98c622003cf83ddbf85527f322b014dba8767c0b5b19965bb44d2263e64be
Image sha256:3f87506b8efebd3c2a82c80a86215c27468ad27fe83b3f49b8d00f6957038f16
LogPath /var/lib/docker/containers/e4e98c622003cf83ddbf85527f322b014dba8767c0b5b19965bb44d2263e64be/e4e98c622003cf83ddbf85527f322b014dba8767c0b5b19965bb44d2263e64be-json.log
MountLabel
Mounts
0 { Destination: /dir16, Driver: local, Mode: z, Name: baa10d52f872581f8251a6f5e2452075e922620b546d28741b960e74565ae842, Propagation: , RW: true, Source: /var/lib/docker/volumes/baa10d52f872581f8251a6f5e2452075e922620b546d28741b960e74565ae842/_data, Type: volume }
1 { Destination: /dir11, Driver: local, Mode: z, Name: cfbeff9562a7b097bccbb4a4871e78b68c779dc5fdf1356b4106c0d6fbeead18, Propagation: , RW: true, Source: /var/lib/docker/volumes/cfbeff9562a7b097bccbb4a4871e78b68c779dc5fdf1356b4106c0d6fbeead18/_data, Type: volume }
2 { Destination: /config, Mode: , Propagation: rprivate, RW: true, Source: /home/floppik/inotify-config, Type: bind }
3 { Destination: /dir8, Driver: local, Mode: z, Name: 758c785b5e42e41284fc67b584e43ebde540c7712b4a511b3876b4151c257890, Propagation: , RW: true, Source: /var/lib/docker/volumes/758c785b5e42e41284fc67b584e43ebde540c7712b4a511b3876b4151c257890/_data, Type: volume }
4 { Destination: /dir17, Driver: local, Mode: z, Name: 3ef3c20c0a61521fc1e3d767d7896b7c3c8c7a0a01d4cdae0289ebfa52691663, Propagation: , RW: true, Source: /var/lib/docker/volumes/3ef3c20c0a61521fc1e3d767d7896b7c3c8c7a0a01d4cdae0289ebfa52691663/_data, Type: volume }
5 { Destination: /dir18, Driver: local, Mode: z, Name: 180f3c1164ab21aa4a49b19543e9611f21de65ec36eb28d01a045f59eaeb7aea, Propagation: , RW: true, Source: /var/lib/docker/volumes/180f3c1164ab21aa4a49b19543e9611f21de65ec36eb28d01a045f59eaeb7aea/_data, Type: volume }
6 { Destination: /dir15, Driver: local, Mode: z, Name: 635c2fe987aaa6b276b89bd667f58675cdebf832404a4808ed7bebd3bebf5a60, Propagation: , RW: true, Source: /var/lib/docker/volumes/635c2fe987aaa6b276b89bd667f58675cdebf832404a4808ed7bebd3bebf5a60/_data, Type: volume }
7 { Destination: /etc/localtime, Mode: , Propagation: rprivate, RW: true, Source: /etc/localtime, Type: bind }
8
Destination /var/run/docker.sock
Mode
Propagation rprivate
RW true
Source /var/run/docker.sock
Type bind
9 { Destination: /dir10, Driver: local, Mode: z, Name: b09dd335db93f0cd57531e22f750e67f3a920378c056a4c4593692e6ab1626a9, Propagation: , RW: true, Source: /var/lib/docker/volumes/b09dd335db93f0cd57531e22f750e67f3a920378c056a4c4593692e6ab1626a9/_data, Type: volume }
10 { Destination: /dir2, Driver: local, Mode: z, Name: 57d9a17b9bbd1e63425542bd1113da7db0ba2f1077a376c597a6b929f2712292, Propagation: , RW: true, Source: /var/lib/docker/volumes/57d9a17b9bbd1e63425542bd1113da7db0ba2f1077a376c597a6b929f2712292/_data, Type: volume }
11 { Destination: /dir4, Driver: local, Mode: z, Name: cc5c474dc43718f237239ef7b5b3c9f0b8185030a1e4358eeb8dcdb9194fd968, Propagation: , RW: true, Source: /var/lib/docker/volumes/cc5c474dc43718f237239ef7b5b3c9f0b8185030a1e4358eeb8dcdb9194fd968/_data, Type: volume }
12 { Destination: /dir13, Driver: local, Mode: z, Name: 1531d2dc17d569157833636302dd2ecb049270300bfba36e15ca429e5ded3893, Propagation: , RW: true, Source: /var/lib/docker/volumes/1531d2dc17d569157833636302dd2ecb049270300bfba36e15ca429e5ded3893/_data, Type: volume }
13 { Destination: /dir9, Driver: local, Mode: z, Name: d0456f96284f3147e785df9c7cbcc2e898a72f7284da18d21a6d3582c8e6e9f1, Propagation: , RW: true, Source: /var/lib/docker/volumes/d0456f96284f3147e785df9c7cbcc2e898a72f7284da18d21a6d3582c8e6e9f1/_data, Type: volume }
14 { Destination: /dir5, Driver: local, Mode: z, Name: 9a72376441359f621fe9dbd0438607a6350376524b94c9a40ebcafcdc8714f00, Propagation: , RW: true, Source: /var/lib/docker/volumes/9a72376441359f621fe9dbd0438607a6350376524b94c9a40ebcafcdc8714f00/_data, Type: volume }
15 { Destination: /dir19, Driver: local, Mode: z, Name: 64bd97a71fa6b083b77ed1074303da117449e6e1b71dfbeec9eaf70b912630cf, Propagation: , RW: true, Source: /var/lib/docker/volumes/64bd97a71fa6b083b77ed1074303da117449e6e1b71dfbeec9eaf70b912630cf/_data, Type: volume }
16 { Destination: /dir1, Mode: , Propagation: rprivate, RW: true, Source: /mnt/Toshiba/nc_data/admin/files/Фото/Автозагрузка/2022, Type: bind }
17 { Destination: /dir7, Driver: local, Mode: z, Name: 5e3889cc7aa5bcc2a29b3badea6ea23413f1037f169506a605758298ed8656c0, Propagation: , RW: true, Source: /var/lib/docker/volumes/5e3889cc7aa5bcc2a29b3badea6ea23413f1037f169506a605758298ed8656c0/_data, Type: volume }
18 { Destination: /dir12, Driver: local, Mode: z, Name: 45d99435e3c2176958fafa507e59ccb78697049b4c5e818ad0dbe7f5cb8005fb, Propagation: , RW: true, Source: /var/lib/docker/volumes/45d99435e3c2176958fafa507e59ccb78697049b4c5e818ad0dbe7f5cb8005fb/_data, Type: volume }
19 { Destination: /dir20, Driver: local, Mode: z, Name: 7b9acb904058c50412b6dd4179177333407e652d24ac5e2bd80a123a80b11f64, Propagation: , RW: true, Source: /var/lib/docker/volumes/7b9acb904058c50412b6dd4179177333407e652d24ac5e2bd80a123a80b11f64/_data, Type: volume }
20 { Destination: /dir6, Driver: local, Mode: z, Name: f74f0a8fd52790cc435242d6851de01f6402023536cde568733e73b818b8e44a, Propagation: , RW: true, Source: /var/lib/docker/volumes/f74f0a8fd52790cc435242d6851de01f6402023536cde568733e73b818b8e44a/_data, Type: volume }
21 { Destination: /dir14, Driver: local, Mode: z, Name: a1bfe0b4fc6f821efd6db3646db1a8ef1c52fe16885feefb05a970673cd71dbf, Propagation: , RW: true, Source: /var/lib/docker/volumes/a1bfe0b4fc6f821efd6db3646db1a8ef1c52fe16885feefb05a970673cd71dbf/_data, Type: volume }
22 { Destination: /dir3, Driver: local, Mode: z, Name: 77d50377090096c9226d743f031ac9023285551bd3aef706aa3ac4c3f55ade6c, Propagation: , RW: true, Source: /var/lib/docker/volumes/77d50377090096c9226d743f031ac9023285551bd3aef706aa3ac4c3f55ade6c/_data, Type: volume }
23
Destination /usr/bin/docker
Mode
Propagation rprivate
RW true
Source /usr/bin/docker
Type bind
Name /inotify-command
NetworkSettings { Bridge: , EndpointID: 52cd4dbca497edb19f70a10e658f69562bf95870a4f9704c07be1d2a305b9f46, Gateway: 172.17.0.1, GlobalIPv6Address: , GlobalIPv6PrefixLen: 0, HairpinMode: false, IPAddress: 172.17.0.8, IPPrefixLen: 16, IPv6Gateway: , LinkLocalIPv6Address: , LinkLocalIPv6PrefixLen: 0, MacAddress: 02:42:ac:11:00:08, Networks: [object Object], Ports: [object Object], SandboxID: 23fc4fec80dc8a9fb006bd2b4aa95a206fdc5ab40921e0fb6fe29526393996d8, SandboxKey: /var/run/docker/netns/23fc4fec80dc, SecondaryIPAddresses: null, SecondaryIPv6Addresses: null }
Path /sbin/boot.sh
Platform linux
Portainer { ResourceControl: [object Object] }
ProcessLabel
ResolvConfPath /var/lib/docker/containers/e4e98c622003cf83ddbf85527f322b014dba8767c0b5b19965bb44d2263e64be/resolv.conf
RestartCount 0
State { Dead: false, Error: , ExitCode: 0, FinishedAt: 0001-01-01T00:00:00Z, OOMKilled: false, Paused: false, Pid: 3289941, Restarting: false, Running: true, StartedAt: 2022-10-03T11:34:31.341913908Z, Status: running }
```

**Comment by flurmind** [comment]
Hi. Thank's for idea! I use nextcloud too for sync (I added my "foto folder" in Nextcloud docker-compose and then attached like external storage and added this folder in photoprism docker-compose like "originals") and couldn't find solution 1 month))
But now I did this without docker on debian:

`#!/bin/bash

log='/srv/disk1/log'
LTIME=`stat -c %Z $log`

inotifywait -m -r -o/srv/disk1/log  --timefmt '%F %T' --format '%T %w%f %e' -e delete,create /srv/disk1/Foto |
while true; do
   ATIME=`stat -c %Z $log`
   if [[ "$ATIME" != "$LTIME" ]]
   then 
       docker exec photoprism photoprism index --cleanup
       LTIME=$ATIME
   fi
   sleep 60
done`

Maybe someone has an idea how to do without a log file  but with interval?

**Comment by burjuyz** [comment]
> The docker I am using is https://hub.docker.com/r/coppit/inotify-command. It uses inotify API to monitor file system events. There's sample configurations and detailed documentation for parameters on the page. The command for indexing is pretty standard: `docker exec photoprism photoprism index /nextcloud`.

COMMAND="docker exec photoprism photoprism index"
My Log is next:
[2022-11-21 15:18:24] Photoprism: Detected change to file /dir1/v9-K41baGwM.jpg
[2022-11-21 15:18:24] Photoprism: Waiting for watch directory to stabilize for 5 seconds before triggering command
[2022-11-21 15:18:29] Photoprism: Watch directory stabilized for 5 seconds. Triggering command.
[2022-11-21 15:18:29] Photoprism: Running command with user ID 0, group ID 0, and umask 0022
[2022-11-21 15:18:29] Photoprism: vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
[2022-11-21 15:18:29] Found existing user "root" with the proper user ID and group ID. Skipping creation of user and group...
[2022-11-21 15:18:29] Running command as user "root"...
setuser: cannot execute docker: [Errno 2] No such file or directory
[2022-11-21 15:18:29] Photoprism: ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2022-11-21 15:18:29] Photoprism: Finished running command. Exit code was 1
[2022-11-21 15:18:29] Photoprism: Waiting for new change
[2022-11-21 15:22:47] Photoprism: Detected change to file /dir1/v9-K41baGwM.jpg
[2022-11-21 15:22:47] Photoprism: Waiting for watch directory to stabilize for 5 seconds before triggering command
[2022-11-21 15:22:52] Photoprism: Watch directory stabilized for 5 seconds. Triggering command.
[2022-11-21 15:22:52] Photoprism: Command triggered, but it's too soon to run the command again. Waiting another 336 seconds
[2022-11-21 15:28:29] Photoprism: Running command with user ID 0, group ID 0, and umask 0022
[2022-11-21 15:28:29] Photoprism: vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
[2022-11-21 15:28:29] Found existing user "root" with the proper user ID and group ID. Skipping creation of user and group...
[2022-11-21 15:28:29] Running command as user "root"...
setuser: cannot execute docker: [Errno 2] No such file or directory
[2022-11-21 15:28:29] Photoprism: ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2022-11-21 15:28:29] Photoprism: Finished running command. Exit code was 1
[2022-11-21 15:28:29] Photoprism: Waiting for new change
[2022-11-21 14:00:36] Photoprism: Waiting for new change
run: Photoprism: (pid 28) 5s
[2022-11-21 14:00:54] Photoprism: Detected change to file /dir1/v9-K41baGwM.jpg
[2022-11-21 14:00:54] Photoprism: Waiting for watch directory to stabilize for 5 seconds before triggering command
[2022-11-21 14:00:59] Photoprism: Watch directory stabilized for 5 seconds. Triggering command.
[2022-11-21 14:00:59] Photoprism: Running command with user ID 100, group ID 1000, and umask 0022
[2022-11-21 14:00:59] Photoprism: vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
[2022-11-21 14:00:59] Creating user "user_100_1000" (ID 100) and group "group_100_1000" (ID 1000) to run the command...
/files/runas.sh: line 82: groupadd: command not found
/files/runas.sh: line 83: useradd: command not found
chown: unknown user/group user_100_1000:group_100_1000
[2022-11-21 14:00:59] Running command as user "user_100_1000"...
setuser: user user_100_1000 not found
[2022-11-21 14:00:59] Photoprism: ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2022-11-21 14:00:59] Photoprism: Finished running command. Exit code was 1

[whoshoe] Can you help me please?

**Comment by burjuyz** [comment]
Any update?

`[2022-11-22 11:19:18] Photoprism: CONFIGURATION:
[2022-11-22 11:19:18] Photoprism: WATCH_DIR=/dir1
[2022-11-22 11:19:18] Photoprism: SETTLE_DURATION=5
[2022-11-22 11:19:18] Photoprism: MAX_WAIT_TIME=300
[2022-11-22 11:19:18] Photoprism: MIN_PERIOD=600
[2022-11-22 11:19:18] Photoprism: COMMAND=docker exec photoprism photoprism index --cleanup
[2022-11-22 11:19:18] Photoprism: USER_ID=0
[2022-11-22 11:19:18] Photoprism: GROUP_ID=0
[2022-11-22 11:19:18] Photoprism: UMASK=0000
[2022-11-22 11:19:18] Photoprism: DEBUG=False
[2022-11-22 11:19:18] Photoprism: USE_POLLING=False
[2022-11-22 11:19:18] Photoprism: IGNORE_EVENTS_WHILE_COMMAND_IS_RUNNING=True
[2022-11-22 11:19:18] Photoprism: Starting monitor for Photoprism
[2022-11-22 11:19:18] Photoprism: Using native change detection to detect changes
[2022-11-22 11:19:18] Photoprism: Waiting for new change
run: Photoprism: (pid 43) 5s
[2022-11-22 11:20:54] Photoprism: Detected change to file /dir1/v9-K41baGwM.jpg
[2022-11-22 11:20:54] Photoprism: Waiting for watch directory to stabilize for 5 seconds before triggering command
[2022-11-22 11:20:59] Photoprism: Watch directory stabilized for 5 seconds. Triggering command.
[2022-11-22 11:20:59] Photoprism: Running command with user ID 0, group ID 0, and umask 0000
[2022-11-22 11:20:59] Photoprism: vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
[2022-11-22 11:20:59] Found existing user "root" with the proper user ID and group ID. Skipping creation of user and group...
[2022-11-22 11:20:59] Running command as user "root"...
setuser: cannot execute docker: [Errno 2] No such file or directory
[2022-11-22 11:20:59] Photoprism: ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2022-11-22 11:20:59] Photoprism: Finished running command. Exit code was 1
[2022-11-22 11:20:59] Photoprism: Waiting for new change`

If I enter the inotify-command container and run the command, the answer is
```
bash-4.4# docker exec photoprism photoprism index --cleanup
bash: /usr/bin/docker: No such file or directory
bash-4.4#
```
I've done before:
map the host docker socket to the container: -v /var/run/docker.sock:/var/run/docker.sock
map the host docker bin file to the container: -v /usr/bin/docker:/usr/bin/docker
